internal/v5/archive.go

```diff
@@ -311,9 +312,10 @@ func (h *ReqHandler) ServeBlobFile(w http.ResponseWriter, req *http.Request, id
 }
 
 func (h *ReqHandler) isPublic(id *router.ResolvedURL) bool {
-	baseEntity, err := h.Cache.BaseEntity(&id.URL, charmstore.FieldSelector("acls"))
+	baseEntity, err := h.Cache.BaseEntity(&id.URL, charmstore.FieldSelector("channelacls"))
 	if err == nil {
-		for _, p := range baseEntity.ACLs.Read {
+		// TODO choose appropriate ACLs based on entity.
```

Comment:
Isn't it as simple as calling h.entityACLs at this point?
Mayeb we should just mention that in the comment...